File: design-pattern-java/app/src/main/java/design/pattern/java/patterns/creational/abstract_factory_method/FordFactory.java
package design.pattern.java.patterns.creational.abstract_factory_method;

public class FordFactory implements VehicleFactory {

  @Override
  public DeportVehicle createDeportCar() {
    return new FordMustang();
  }

  @Override
  public RusticVehicle createRusticCar() {
    return new FordPickup();
  }
  
}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/creational/abstract_factory_method/FordFactory.kt
package design.pattern.kotlin.patterns.creational.abstract_factory_method

class FordFactory : VehicleFactory {
    override fun createDeportCar(): DeportVehicle {
        return FordMustang()
    }

    override fun createRusticCar(): RusticVehicle {
        return FordPickup()
    }
}